--- wiz_socket.h	2020-12-01 17:45:13.474682643 +0900
+++ wiz_socket.h	2020-11-30 10:44:03.145529792 +0900
@@ -82,8 +82,8 @@
  * Following figure shows network flow diagram by WIZnet SOCKET API.
  * @image html WIZnet_SOCKET.jpg "<WIZnet SOCKET API>"
  */
-#ifndef _SOCKET_H_
-#define _SOCKET_H_
+#ifndef _WIZ_SOCKET_H_
+#define _WIZ_SOCKET_H_
 #ifdef __cplusplus
  extern "C" {
 #endif
@@ -162,7 +162,7 @@
  *                        @ref SOCKERR_SOCKMODE    - Not support socket mode as TCP, UDP, and so on. \n
  *                        @ref SOCKERR_SOCKFLAG    - Invaild socket flag.
  */
-int8_t  socket(uint8_t sn, uint8_t protocol, uint16_t port, uint8_t flag);
+int8_t  wiz_socket(uint8_t sn, uint8_t protocol, uint16_t port, uint8_t flag);
 
 /**
  * @ingroup WIZnet_socket_APIs
@@ -174,7 +174,7 @@ int8_t  socket(uint8_t sn, uint8_t proto
  * @return @b Success : @ref SOCK_OK \n
  *         @b Fail    : @ref SOCKERR_SOCKNUM - Invalid socket number
  */
-int8_t  close(uint8_t sn);
+int8_t  wiz_close(uint8_t sn);
 
 /**
  * @ingroup WIZnet_socket_APIs
@@ -187,7 +187,7 @@ int8_t  close(uint8_t sn);
  *         @b Fail    :\n @ref SOCKERR_SOCKINIT   - Socket is not initialized \n
  *                        @ref SOCKERR_SOCKCLOSED - Socket closed unexpectedly.
  */
-int8_t  listen(uint8_t sn);
+int8_t  wiz_listen(uint8_t sn);
 
 /**
  * @ingroup WIZnet_socket_APIs
@@ -210,7 +210,7 @@ int8_t  listen(uint8_t sn);
  *                @ref SOCKERR_TIMEOUT   - Timeout occurred during request connection\n
  *                @ref SOCK_BUSY         - In non-block io mode, it returned immediately\n
  */
-int8_t  connect(uint8_t sn, uint8_t * addr, uint16_t port);
+int8_t  wiz_connect(uint8_t sn, uint8_t * addr, uint16_t port);
 
 /**
  * @ingroup WIZnet_socket_APIs
@@ -227,7 +227,7 @@ int8_t  connect(uint8_t sn, uint8_t * ad
  *                        @ref SOCKERR_TIMEOUT  - Timeout occurred \n
  *                        @ref SOCK_BUSY        - Socket is busy.
  */
-int8_t  disconnect(uint8_t sn);
+int8_t  wiz_disconnect(uint8_t sn);
 
 /**
  * @ingroup WIZnet_socket_APIs
@@ -247,7 +247,7 @@ int8_t  disconnect(uint8_t sn);
  *                          @ref SOCKERR_DATALEN    - zero data length \n
  *                          @ref SOCK_BUSY          - Socket is busy.
  */
-int32_t send(uint8_t sn, uint8_t * buf, uint16_t len);
+int32_t wiz_send(uint8_t sn, uint8_t * buf, uint16_t len);
 
 /**
  * @ingroup WIZnet_socket_APIs
@@ -269,7 +269,7 @@ int32_t send(uint8_t sn, uint8_t * buf, 
  *                     @ref SOCKERR_DATALEN    - zero data length \n
  *                     @ref SOCK_BUSY          - Socket is busy.
  */
-int32_t recv(uint8_t sn, uint8_t * buf, uint16_t len);
+int32_t wiz_recv(uint8_t sn, uint8_t * buf, uint16_t len);
 
 /**
  * @ingroup WIZnet_socket_APIs
@@ -297,7 +297,7 @@ int32_t recv(uint8_t sn, uint8_t * buf, 
  *                        @ref SOCKERR_TIMEOUT     - Timeout occurred \n
  *                        @ref SOCK_BUSY           - Socket is busy. 
  */
-int32_t sendto(uint8_t sn, uint8_t * buf, uint16_t len, uint8_t * addr, uint16_t port);
+int32_t wiz_sendto(uint8_t sn, uint8_t * buf, uint16_t len, uint8_t * addr, uint16_t port);
 
 /**
  * @ingroup WIZnet_socket_APIs
@@ -327,7 +327,7 @@ int32_t sendto(uint8_t sn, uint8_t * buf
  *                       @ref SOCKERR_SOCKNUM    - Invalid socket number \n
  *                       @ref SOCKBUSY           - Socket is busy.
  */
-int32_t recvfrom(uint8_t sn, uint8_t * buf, uint16_t len, uint8_t * addr, uint16_t *port);
+int32_t wiz_recvfrom(uint8_t sn, uint8_t * buf, uint16_t len, uint8_t * addr, uint16_t *port);
 
 
 /////////////////////////////
@@ -418,7 +418,7 @@ typedef enum
  *  @return @b Success @ref SOCK_OK \n
  *          @b fail    @ref SOCKERR_ARG         - Invalid argument\n
  */
-int8_t  ctlsocket(uint8_t sn, ctlsock_type cstype, void* arg);
+int8_t  wiz_ctlsocket(uint8_t sn, ctlsock_type cstype, void* arg);
 
 /** 
  * @ingroup WIZnet_socket_APIs
@@ -446,7 +446,7 @@ int8_t  ctlsocket(uint8_t sn, ctlsock_ty
  *  - @ref SOCKERR_SOCKOPT     - Invalid socket option or its value \n
  *  - @ref SOCKERR_TIMEOUT     - Timeout occurred when sending keep-alive packet \n
  */
-int8_t  setsockopt(uint8_t sn, sockopt_type sotype, void* arg);
+int8_t  wiz_setsockopt(uint8_t sn, sockopt_type sotype, void* arg);
 
 /** 
  * @ingroup WIZnet_socket_APIs
@@ -480,10 +480,10 @@ int8_t  setsockopt(uint8_t sn, sockopt_t
  *   When SO_PACKINFO value is PACK_FIRST and the return value of recvfrom() is zero, 
  *   This means the zero byte UDP data(UDP Header only) received.
   */
-int8_t  getsockopt(uint8_t sn, sockopt_type sotype, void* arg);
+int8_t  wiz_getsockopt(uint8_t sn, sockopt_type sotype, void* arg);
 
 #ifdef __cplusplus
  }
 #endif
 
-#endif   // _SOCKET_H_
+#endif   // _WIZ_SOCKET_H_
